app/Http/Controllers/AlbumsController.php

```diff
@@ -70,78 +72,29 @@ public function getPositionData()
 
 		// Initialize return var
 		$return = array();
-		$return['albums'] = array();
 
-		// Get photos
-		// Get album information
-		$UserId = $this->sessionFunctions->id();
-		$full_photo = Configs::get_value('full_photo', '1') == '1';
-
-		$toplevel = $this->albumFunctions->getToplevelAlbums();
-		if ($toplevel === null) {
-			return Response::error('I could not find you.');
-		}
-
-		// All normal albums
-		$toplevel_albums = $this->albumFunctions->prepare_albums($toplevel['albums']);
-		// Add albums shared with users
-		$toplevel_albums = array_merge($toplevel_albums, $this->albumFunctions->prepare_albums($toplevel['shared_albums']));
-
-		$toplevel_album_ids = array();
-		$album_list = array();
-
-		foreach ($toplevel_albums as $album_iterator) {
-			$toplevel_album_ids[] = $album_iterator['id'];
-
-			$album_tmp = Album::with('children')->find($album_iterator['id']);
-			$full_photo = $album_tmp->full_photo_visible();
+		$albumIDs = $this->albumFunctions->getPublicAlbums();
+
+		$query = Photo::with('album')->where(
+				function (Builder $query) use ($albumIDs) {
```

Comment:
Hummm I am not happy with this function. There should be a simpler way to achieve that.